wordpressをdockerfileに移行

matsumoto-ops opened the pull request:
## 背景
https://github.com/matsumoto-ops/menta/pull/1
で作成したdockercomposeからwordpress部分をDockerfileに切り出し


## 変更内容
1. docker-compose.ymlからwordpress部分を削除
1. dockerfileを追加してwordpress部分を追記

## issues
#3

## 確認手順
1. `docker-compose up -d` でコンテナ立ち上げ
1. localhost:8000でブラウザアクセスしてwordpressが起動することを確認

![image](https://user-images.githubusercontent.com/55909058/87875852-1abaee80-ca0f-11ea-82eb-39de19dd7193.png)

inamuu commented on wordpress/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+#RUN systemctl enable httpd 
+
+# 接続用ポートの穴あけ
+EXPOSE 8010
```
ここで8010を指定していますが、あくまでも受付ポートなので実際はApacheはデフォルトで80になっているかと思います。
なので80にしておいたほうが良いかもです。
docker-compose.ymlのport指定で8010:80のようにすることで、ポートフォワードができるようになります。

inamuu commented on wordpress/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+
+# yumのアップデート
+RUN yum update -y &&\
+    yum clean all
```
`yum clean all`は最後に一回やるだけで大丈夫なのと、PHPのライブラリインストールしている際にyum cleanしていないので、その分不要なファイルが残っているのでそのあとにyum clean allをしてください、

inamuu commented:
@matsumoto-ops 
コンフリクトを起こしているのでこちらを解消してください。
恐らく、前のブランチをpullしないでブランチを切ってしまっているので、このブランチの設定を反映するように解消してみてください。

matsumoto-ops commented:
一旦ごちゃごちゃしたのでwordpressブランチを新しく切って
そこに修正をpushしました。
ただ過去に作成したmysqlフォルダが残ってしまってます。
一旦リモートブランチをローカルに強制反映させて
mysqlフォルダを削除 -> add -> commit -pushしたのですが
削除が反映されません。

https://www.zunouissiki.com/entry/cannot-git-push-everything-up-to-date/
サイトを見るとコミットのし忘れ見たいに書いてあるのですが、コミットは
履歴を見ても実施しているはずなのですが、原因が分からず。



```
$ git fetch origin 
remote: Enumerating objects: 13, done.
remote: Counting objects: 100% (13/13), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 11 (delta 1), reused 6 (delta 1), pack-reused 0
Unpacking objects: 100% (11/11), 1.59 KiB | 6.00 KiB/s, done.
From https://github.com/matsumoto-ops/menta
   23fb1ae..bdfacb2  master     -> origin/master

admin@home-pc MINGW64 ~/OneDrive/work/menta/wordpress/menta (wordpress)
$ git reset --hard origin/wordpress 
HEAD is now at 8e9185e chenge dockerfile

admin@home-pc MINGW64 ~/OneDrive/work/menta/wordpress/menta (wordpress)
$ git add *

admin@home-pc MINGW64 ~/OneDrive/work/menta/wordpress/menta (wordpress)
$ git status
On branch wordpress
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        deleted:    mysql/Dockerfile

no changes added to commit (use "git add" and/or "git commit -a")

admin@home-pc MINGW64 ~/OneDrive/work/menta/wordpress/menta (wordpress)
$ git commit -m "del mysql"
On branch wordpress
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        deleted:    mysql/Dockerfile

no changes added to commit (use "git add" and/or "git commit -a")

admin@home-pc MINGW64 ~/OneDrive/work/menta/wordpress/menta (wordpress)
$ git push origin wordpress 
Everything up-to-date
```

matsumoto-ops commented:
>ただ過去に作成したmysqlフォルダが残ってしまってます。

↑は解決しました。
✕ git add *
〇 git add a
でした。
いままで何も考えず`git add *`に
で行っていたんですが、サブフォルダは対象にならないんですね。

inamuu commented:
@matsumoto-ops 
ブランチは一つでやってみたほうが良いかもです。
個人でやっている範囲では、それでOKなのですがチーム開発でやっている際に、自分のミスの度にブランチを作り直しているとレビュワーが混乱してしまうからです。
今レビューした内容は新しいブランチには反映されないので、レビューした箇所が修正されたかどうかを、また1から見直す必要が発生してしまいます。
GitHubは指摘した箇所が修正されれば、それがパッと見でわかるのでレビュワーにも優しい設計になっている点も、チーム開発には重要なポイントとなります。
いずれはチーム開発を想定されているのかと思いますので、是非意識していただけると良さそうです。
1つのブランチでごちゃごちゃしてしまっても、元に戻せるのがgitの良いところなので、頑張ってやってみましょう💪

ちなみにですが、
```
$ git commit -m "del mysql"
On branch wordpress
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        deleted:    mysql/Dockerfile

no changes added to commit (use "git add" and/or "git commit -a")
```
これを見てみると、`Changes not staged for commit:` となり、 `no changes added to commit (use "git add" and/or "git commit -a")` と表示されていますね。
なので、恐らくファイルをrmしたあとに、addしていないのでは無いでしょうか。
gitの場合、今ファイルがどの状態にあるかは意識すると良いですね。

今回の場合、AブランチとBブランチをわけた際に、Aで作成したファイルがBで残ってしまうのは、Aブランチでファイルを削除していなければ、Bブランチに切り替えてもファイルは削除されないません。
もちろんAブランチでファイルを削除していればBランチに切り替えた際に復活することはありません。

untracked fileとなった場合は下記のようにやることも出来ます。
https://qiita.com/tmyn470/items/c8359e4ec92d1f462bdf

あとサブフォルダもまとめてやる場合、私はピリオドでカレントディレクトリを指定しています。
こんな方法もあるので参考にしてみてください。
```sh
git add .
```

inamuu commented:
どこかのコミットのタイミングまで戻したい場合は、
git logで過去のコミットハッシュ値を確認して、
git reset --hard コミットハッシュ値にして戻し、
新たにコミットしたら、
git push -f ブランチ名
で強制的にpushしてあげることで、上書きすることができます。

上記やり取りは、git reflog で確認することが可能です。
